### ISSUE TITLE
maven2 repository type not accepted.

### ISSUE THREAD

**Issue by c4rlier** [datadrivers/terraform-provider-nexus#37]
Hi,
Firstly thank you for solving the previous issue regarding maven repositories. There is another issue that came up during exploratory testing.

The provider accepts `maven` as a repository type, but it should accept `maven2`.  When maven 2 is input you see the error in code block 1). when you change it to `maven` the tf code executes just fine, but it creates maven2 repositories in nexus (when you insepct the UI). On a subsequent tf apply it tries to change the repo type from `maven2` to `maven`. tf code can be found in block 0).

blocks:
0)
```
resource "nexus_repository" "maven_hosted" {
  depends_on = [helm_release.nexus]

  name   = "maven-hosted"
  format = "maven"
  type   = "hosted"

  maven {
    version_policy = "MIXED"
    layout_policy  = "STRICT"
  }


  storage {
    blob_store_name                = nexus_blobstore.s3.name
    strict_content_type_validation = true
    write_policy                   = "ALLOW"
  }
}
```

1)
```
Warning: This resource is deprecated and will be removed in the next major version.
Please supply the URL of your repository to helm_release resources directly, using the repository attribute.
See: https://www.terraform.io/docs/providers/helm/r/release.html#example-usage

  on modules/helm-nexus/main.tf line 61, in data "helm_repository" "oteemo":
  61: data "helm_repository" "oteemo" {



Error: expected format to be one of [apt bower conan docker helm maven npm nuget pypi yum], got maven2

  on modules/helm-nexus/nexus_repositories_maven.tf line 82, in resource "nexus_repository" "maven_hosted":
  82: resource "nexus_repository" "maven_hosted" {


make: *** [apply] Error 1
```

2)
```
  # module.helm_nexus.nexus_repository.maven_hosted must be replaced
-/+ resource "nexus_repository" "maven_hosted" {
      ~ format = "maven2" -> "maven" # forces replacement
      ~ id     = "maven-hosted" -> (known after apply)
        name   = "maven-hosted"
        online = true
        type   = "hosted"

        maven {
            layout_policy  = "STRICT"
            version_policy = "MIXED"
        }

        storage {
            blob_store_name                = "REDACTED"
            strict_content_type_validation = true
            write_policy                   = "ALLOW"
        }
    }
```

**Comment by Nosmoht** [comment]
Hi @c4rlier ,

we just identified the issue and will fix it soon.

**Comment by Nosmoht** [comment]
Fixed with release [v1.5.0](https://github.com/datadrivers/terraform-provider-nexus/releases/tag/v1.5.0)

**Comment by Nosmoht** [comment]
@c4rlier could you please test with v1.5.3 and let us know the result?
